### DISCUSSION TITLE
mpd 0.23: switch from libupnp to libnpupnp

### DISCUSSION THREAD

**Discussion by aeolio** [MusicPlayerDaemon/MPD#1348] [Q&A] [answered]
Hi,

Just a question: What was the reason behind the decision to switch UPnP libraries? Is libnpupnp more stable than the original version? The changelog is not very communicative :-D

Kind regards,
Andreas

**Comment by MaxKellermann** [comment, marked as answer]
There was no decision to switch libraries. Both are supported.

**Comment by aeolio** [comment]
And is there a recommended library to use for UPnP integration?

**Reply by MaxKellermann** [reply]
I can't recommend one. I use neither.
I didn't write the UPnP code - I only fixed lots of bugs in there, and found libupnp to be a very badly designed library. I don't know about libnpupnp, but I don't think it's any better.

**Comment by aeolio** [comment]
I was not complaining ... 
Thank you for your time and the explanation!
Kind regards,
Andreas
